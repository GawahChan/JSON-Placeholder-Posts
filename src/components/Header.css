.header {
    text-align: center;
    background-color: #415E7B;
    padding: 1rem 1rem;
    color: #DDDDDD;
}

.header h1 {
    font-weight: 400;
}

.header h3 {
    font-weight: 400;
}

.input-filterPosts {
    margin: 0.5rem;
    padding: 10px;
    width: 200px;
}