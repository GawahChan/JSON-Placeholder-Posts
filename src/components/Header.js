import React from 'react';

function Header(props) {
    return (
        <div>
            <h1>{props.title}</h1>
            <input
                className='input-filterPosts'
                value={props.inputValue}
                placeholder='Please select between 1 and 10'
                onChange={(inputValue) => props.filterPosts(inputValue.target.value)} 
                />
        </div>
    );
}

export default Header;