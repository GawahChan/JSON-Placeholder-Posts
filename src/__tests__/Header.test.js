import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Header from '../components/Header';

describe('Header', () => {
    const mockInput = jest.fn();
    const mockInputChange = 3;
    const headerTitle = 'Posts Page'
    const props = { title: headerTitle, filterPosts: mockInput }

    const headerTree = renderer.create(<Header {...props} />).toJSON();
    const header = shallow(<Header {...props} />);

    it('renders correctly', () => {
        expect(headerTree).toMatchSnapshot();
    });

    it('calls `filterPosts` callback', () => {
        header.find('.input-filterPosts').simulate('change', {target: { value: mockInputChange }});

        expect(mockInput).toHaveBeenCalledWith(mockInputChange);
    })
})