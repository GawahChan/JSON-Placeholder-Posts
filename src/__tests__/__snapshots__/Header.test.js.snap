// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div>
  <h1>
    Posts Page
  </h1>
  <input
    className="input-filterPosts"
    onChange={[Function]}
    placeholder="Please select between 1 and 10"
  />
</div>
`;
